go-controller/pkg/ovn/base_network_controller_pods.go

```diff
@@ -635,24 +670,31 @@ func (bnc *BaseNetworkController) addLogicalPortToNetwork(pod *kapi.Pod, nadName
 		}
 	}()
 
-	if err == nil {
+	if podAnnotation != nil {
 		podMac = podAnnotation.MAC
 		podIfAddrs = podAnnotation.IPs
 
 		// If the pod already has annotations use the existing static
 		// IP/MAC from the annotation.
 		lsp.DynamicAddresses = nil
 
-		if bnc.doesNetworkRequireIPAM() {
-			// ensure we have reserved the IPs in the annotation
-			if err = bnc.lsManager.AllocateIPs(switchName, podIfAddrs); err != nil && err != ipallocator.ErrAllocated {
-				return nil, nil, nil, false, fmt.Errorf("unable to ensure IPs allocated for already annotated pod: %s, IPs: %s, error: %v",
-					podDesc, util.JoinIPNetIPs(podIfAddrs, " "), err)
-			} else {
-				needsIP = false
+		// At interconnect the logical switch manager only knows stuff about
+		// local zone nodes if we ask for original switch after
+		// live migration this fails, that's why we skip it here, also on
+		// controller restart the IPs are already allocated by syncPods.
+		needsIP = len(podAnnotation.IPs) == 0
+		if needsIP {
```

Comment:
This is not equivalent to what you had before:
```
needsIP := current != original
```
So before you wouldn't try to allocate pods that migrated to a different now.

What yo uhave no is no different than legacy code because `err == nil` or `podAnnotation != nil` is basically equivalent to `len(podAnnotation.IPs) > 0` so here `needsIP == true` is guaranteed after `podAnnotation != nil`, if the annotation is set is guaranteed to have IPs, it does not make sense to set an annotation without IPS (on the default network)

This means you will allocate IPs for migrated pods, even if if on different than original node/zone anf that you will potentially get and report an error on `AllocateIPs`

What I expected here is that you checked if the logical switch manager was managing the IPs and set that bool to needsIP, or something along those lines.

Reply:
The pod annotation has being "ensured" before this code, so any of the checks you mention will work as expected.

Reply:
Ok, it's me reading the code backwards. Sorry for that. 

My concern then it's the other way around as well: for normal non migrated pods, you are not allocating the annotated IPs as the code was doing before. 

This might seem unnecessary but it might act as a safeguard in case syncPods fails. 

I think is better to keep the old functionality as it was in this case. We traditionally have be avoiding unnecessary functional changes here out of fear to break something.

Reply:
Make sense, I will keep the old code but with a guard for live migrated pod checking that the switch manager contains the pod's expected switch to be sure that it do not fail.